packages/server/src/fhir/repo.ts

```diff
@@ -573,7 +574,11 @@ export class Repository extends FhirRepository<PoolClient> implements Disposable
     }
   }
 
-  private async updateResourceImpl<T extends Resource>(resource: T, create: boolean, versionId?: string): Promise<T> {
+  private async updateResourceImpl<T extends Resource>(
+    resource: T,
+    create: boolean,
+    versionId?: string
+  ): Promise<WithId<T>> {
     const { resourceType, id } = resource;
```

Comment:
This is one place where the `isResourceWithId` narrowing function could be used:

```suggestion
    if (!isResourceWithId(resource)) {
      throw new OperationOutcomeError(badRequest('Missing id'));
    }
    if (!validator.isUUID(resource.id)) {
      throw new OperationOutcomeError(badRequest('Invalid id'));
    }

    const { resourceType, id } = resource;
```

Reply:
+1, this was a suggestion I made in passing yesterday